### DISCUSSION TITLE
Migrate to TypeScript

### DISCUSSION THREAD

**Discussion by nyctfall** [jaredhanson/passport#961] [Ideas]
# Convert Passport.js Core Codebase to TypeScript

Passport.js is a great library for all the underpinnings of authentication, especially for Express apps.

But it could be _better_!

One way it could be improved is with better type-checking and documentation.

I would be willing to make the PR for a complete migration from plain JS to TypeScript for this repo and any other core libs that I have the time for.
And for documentation I would need someone more experienced with this repo and the Passport.js ecosystem for some help on _intended_ usage.

**Comment by jaredhanson** [comment]
I have no interest in converting this package to TypeScript, and would not merge any PR that did so.  PRs that enhance documentation would be welcome, however.

**Reply by nyctfall** [reply]
Thanks for the feedback! I'll start working on some documentation improvements.
